src/main/kotlin/Menu.kt

```diff
@@ -0,0 +1,96 @@
+import java.util.*
+
+interface Menu {
+    val itemMake: String
+    val itemExit: String
+    val intro: String
+    val introInput: String
+    val inputName: String
+    val listItemsMenu: MutableList<String>
+
+
+    fun <K : String, V> prepareListItemMenu(map: MutableMap<K, V?>): MutableList<String> {
+
+        listItemsMenu.clear()
+
+        listItemsMenu.add(itemMake)
+        val listKeysMap = map.keys.toList()
+
+        for (i in listKeysMap) {
+            listItemsMenu.add(i)
+        }
+
+        listItemsMenu.add(itemExit)
+        return listItemsMenu
+    }
+
+
+    fun <K : String, V> listMenu(map: MutableMap<K, V?>, console: Scanner): MutableMap<K, V> {
+        val myNote = Note()
+        val mapNote: MutableMap<String, String?> = mutableMapOf()
+        while (true) {
+            prepareListItemMenu(map)
+            println(intro)
+            for (i in listItemsMenu) {
+                println("${listItemsMenu.indexOf(i)}. $i")
+            }
+            println("\nВыберите пункт меню, для этого введите его номер\n")
+            if (console.hasNextInt()) {
+                val command: Int = console.nextInt()
+                if (command >= 0 && command < listItemsMenu.size) {
+                    when (command) {
+                        listItemsMenu.indexOf(itemMake) -> inputItem(map, console)
+                        listItemsMenu.indexOf("Выход") -> {
+                            println("Выходим на предыдущее меню\n")
+                            break
+                        }
+
+                        else -> {
+                            if (itemMake == "Создать заметку") {
+                                println("\nЗаметка: ${listItemsMenu[command].toUpperCase()}\nСодержание:\n${map[listItemsMenu[command]]}\n")
```

Comment:
🍏 Метод toUpperCase устарел, лучше использовать upperCase
```suggestion
                                println("\nЗаметка: ${listItemsMenu[command].uppercase()}\nСодержание:\n${map[listItemsMenu[command]]}\n")
```